core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -155,7 +160,14 @@ private[spark] class EventLoggingListener(
   }
 
   // Events that do not trigger a flush
-  override def onStageSubmitted(event: SparkListenerStageSubmitted): Unit = logEvent(event)
+  override def onStageSubmitted(event: SparkListenerStageSubmitted): Unit = {
+    logEvent(event)
+    if (shouldLogStageExecutorMetrics) {
+      // record the peak metrics for the new stage
+      liveStageExecutorMetrics.put((event.stageInfo.stageId, event.stageInfo.attemptNumber()),
+        new HashMap[String, ExecutorMetrics]())
```

Comment:
`mutable.Map.empty[String, ExecutorMetrics]` if the above signature is changed.

Reply:
Modified.